l20n/migrate/context.py

```diff
@@ -0,0 +1,210 @@
+# coding=utf8
+
+import os
+import codecs
+import logging
+
+import l20n.format.ast as FTL
+from l20n.format.parser import FTLParser
+from l20n.format.serializer import FTLSerializer
+from compare_locales.parser import getParser
+
+from cldr import get_plural_categories
+from transforms import SOURCE
+from merge import merge_resource
+from util import get_entity, fold_ftl
+
+
+class MergeContext(object):
+    """Stateful context for merging translation resources.
+
+    `MergeContext` must be configured with the target language and the
+    directory locations of the input data.
+
+    The transformation takes four types of input data:
+
+        - The en-US FTL reference files which will be used as templates for
+          message order, comments and sections.
+
+        - The current FTL files for the given language.
+
+        - The legacy (DTD, properties) translation files for the given
+          language.  The translations from these files will be transformed
+          into FTL and merged into the existing FTL files for this language.
+
+        - A list of `FTL.Entity` objects whose some nodes are special operation
```

Comment:
"whose some nodes" ... do we really need to find the grammar book on this one? Reminds of that German story you told me on the streets of Berlin about order of these in German. My Germanic me insists to not mix "whose" and "some".